src/netlink.c

```diff
@@ -387,20 +392,31 @@ static int link_filter_cb(const struct nlmsghdr *nlh,
 
 #if HAVE_TCA_STATS2
 static int qos_attr_cb(const struct nlattr *attr, void *data) {
-  struct gnet_stats_basic **bs = (struct gnet_stats_basic **)data;
+  struct gnet_stats *stats = (struct gnet_stats *)data;
 
   /* skip unsupported attribute in user-space */
   if (mnl_attr_type_valid(attr, TCA_STATS_MAX) < 0)
     return MNL_CB_OK;
 
   if (mnl_attr_get_type(attr) == TCA_STATS_BASIC) {
-    if (mnl_attr_validate2(attr, MNL_TYPE_UNSPEC, sizeof(**bs)) < 0) {
+    if (mnl_attr_validate2(attr, MNL_TYPE_UNSPEC, sizeof(*(stats->bstats))) < 0) {
+      ERROR("netlink plugin: qos_attr_cb: TCA_STATS_BASIC mnl_attr_validate2 "
+            "failed.");
+      return MNL_CB_ERROR;
+    }
+    stats->bstats = mnl_attr_get_payload(attr);
+    if (stats->qstats != NULL)
```

Comment:
This is a bit hard to read, since "bstats" and "qstats" look so similar. How about doing this instead:

```c
if (mnl_attr_get_type(attr) == TCA_STATS_BASIC) {
  // …
  stats->bstats = mnl_attr_get_payload(attr);
} else if (mnl_attr_get_type(attr) == TCA_STATS_QUEUE) {
  // …
  stats->qstats = mnl_attr_get_payload(attr);
}

if ((stats->bstats != NULL) && (stats->qstats != NULL)) {
  return MNL_CB_STOP;
}
return MNL_CB_OK;
```